pkg/autodiscovery/config_poller.go

```diff
@@ -50,14 +50,62 @@ func (pd *configPoller) stop() {
 }
 
 // start starts polling the provider descriptor
-func (pd *configPoller) start(ac *AutoConfig) {
+func (pd *configPoller) start(ctx context.Context, ac *AutoConfig) {
+	_, isStreaming := pd.provider.(providers.StreamingConfigProvider)
+
+	if !isStreaming {
+		pd.pollOnce(ctx, ac)
```

Comment:
Should this be below `canPoll` check?

Reply:
I'll rename `pollOnce` to `collectOnce` to avoid confusion. All providers need a first collection to be done when starting so after `LoadAndRun` all available configs are loaded. This is because of `agent check`. So this `collectOnce` needs to be called by all non-streaming providers, even the ones that don't poll after they start (like the `file` provider)